pkg/ddl/index.go

```diff
@@ -204,6 +204,14 @@ func checkIndexColumn(ctx sessionctx.Context, col *model.ColumnInfo, indexColumn
 		return errors.Trace(dbterror.ErrJSONUsedAsKey.GenWithStackByArgs(col.Name.O))
 	}
 
+	// Vector column cannot index, for now.
+	if col.FieldType.GetType() == mysql.TypeTiDBVectorFloat32 {
+		if col.Hidden {
+			return dbterror.ErrFunctionalIndexOnJSONOrGeometryFunction
```

Comment:
It should be another defined error or use `errors.Errorf(xxx)`

Reply:
I modify it into `return errors.Errorf("ErrFunctionalIndexOnVectorFloat32")`

Reply:
@ericzequan Errorf returns the error message so it should be readable instead of a error code.

Reply:
@breezewish  here i changed into `return errors.Errorf("Functional indexes are not supported on hidden VectorFloat32 columns")`